qa/wildfly/build.gradle

```diff
@@ -73,177 +45,52 @@ dependencies {
   compile "org.apache.logging.log4j:log4j-api:${versions.log4j}"
   compile "org.apache.logging.log4j:log4j-core:${versions.log4j}"
   compile project(path: ':client:rest-high-level', configuration: 'shadow')
-  wildfly "org.jboss:wildfly:${wildflyVersion}@zip"
   testCompile project(':test:framework')
 }
 
-task unzipWildfly(type: Sync) {
-  into wildflyDir
-  from { zipTree(configurations.wildfly.singleFile) }
+war {
+  archiveName "example-app.war"
 }
 
-task deploy(type: Copy) {
-  dependsOn unzipWildfly, war
-  from war
-  into "${wildflyInstall}/standalone/deployments"
-}
-
-task writeElasticsearchProperties(type: DefaultTestClustersTask) {
-  onlyIf { !Os.isFamily(Os.FAMILY_WINDOWS) }
-  useCluster testClusters.integTest
-  dependsOn deploy
-  doLast {
-    final File elasticsearchProperties = file("${wildflyInstall}/standalone/configuration/elasticsearch.properties")
-    elasticsearchProperties.write(
-      [
-        "http.uri=${-> testClusters.integTest.getAllHttpSocketURI().get(0)}"
-      ].join("\n"))
+elasticsearch_distributions {
+  docker {
+    type = 'docker'
+    flavor = System.getProperty('tests.distribution', 'default')
+    version = VersionProperties.getElasticsearch()
+    failIfUnavailable = false // This ensures we skip this testing if Docker is unavailable
   }
 }
 
-// the default configuration ships with IPv6 disabled but our cluster could be bound to IPv6 if the host supports it
-task enableIPv6 {
-  dependsOn unzipWildfly
-  doLast {
-    final File standaloneConf = file("${wildflyInstall}/bin/standalone.conf")
-    final List<String> lines =
-      Files.readAllLines(standaloneConf.toPath())
-        .collect { line -> line.replace("-Djava.net.preferIPv4Stack=true", "-Djava.net.preferIPv4Stack=false") }
-    standaloneConf.write(lines.join("\n"))
-  }
+preProcessFixture {
+  dependsOn elasticsearch_distributions.docker
```

Comment:
We should also depend on `war` here as well, as otherwise there's nothing to actually ensure we build the war artifact before spinning up the docker test fixture. I suspect this is why we are failing in CI [here](https://gradle-enterprise.elastic.co/s/ij4zb4hopk55k/timeline?hideTimeline&taskFilter=:qa:wildfly).

Reply:
Added